src/shared/lib/unit.ts

```diff
@@ -7,7 +7,7 @@ export const getAbsoluteMetricValue = (value: Value, unitId?: UnitId): number |
     return value;
   }
   if (typeof value === 'object') {
-    if (!unitId) {
+    if (!unitId || typeof value[unitId] === 'undefined') {
```

Comment:
crit. dangerous, that may cause a bug in the future (e.g. if user selects a currency which was not returned)